frontends/uhdm/UhdmAst.cc

```diff
@@ -85,16 +85,29 @@ void UhdmAst::visit_default_expr(vpiHandle obj_h)  {
 		auto assign_node = new AST::AstNode(AST::AST_ASSIGN_EQ);
 		auto id_node = new AST::AstNode(AST::AST_IDENTIFIER);
 		id_node->str = current_node->str;
+
+		auto always_child = mod->children.begin();
 		for (auto child : mod->children) {
 			if (child->type == AST::AST_INITIAL) {
 				initial_node = child;
+			}
+
+			if (initial_node != nullptr && (*always_child)->type == AST::AST_ALWAYS)
 				break;
+
+			if ((*always_child)->type != AST::AST_ALWAYS) {
+				always_child++;
 			}
 		}
-		// Ensure single AST_INITIAL node in AST_MODULE
+		// Ensure single AST_INITIAL node is located in AST_MODULE
+		// before any AST_ALWAYS
 		if (initial_node == nullptr) {
 			initial_node = new AST::AstNode(AST::AST_INITIAL);
-			mod->children.push_back(initial_node);
+			if ((*always_child)->type == AST::AST_ALWAYS) {
+				mod->children.insert(always_child, 1, initial_node);
+			} else {
+				mod->children.push_back(initial_node);
+			}
```

Comment:
```suggestion
			if (always_child) {
				mod->children.insert(always_child, 1, initial_node);
			} else {
				mod->children.push_back(initial_node);
			}
```

Reply:
`insert()` requires first argument to be an iterator